tests/integration-tests/utils.py

```diff
@@ -38,6 +38,39 @@
 }
 
 
+class SetupError(BaseException):
+    """Exception to throw from infrastructure factory fixtures if the infrastructure setup fails"""
+
+    def __init__(self, *args, stack_events=None, cluster_details=None):
+        self.message = self._format_message(args[0], stack_events, cluster_details)
+
+    @staticmethod
+    def _format_message(message, stack_events, cluster_details) -> str:
+        formatted_message = message if message else "Error during setup."
+        if cluster_details:
+            details_string = "\n\t".join(
+                [
+                    f"* {failure['failureCode']}:\n\t\t{failure['failureReason']}"
+                    for failure in cluster_details["failures"]
+                ],
+            )
+            formatted_message += f"\n\n- Cluster Errors:\n\t{details_string}"
+
+        if stack_events:
+            events_string = "\n\t".join(
+                [
+                    f"* {event['LogicalResourceId']}:\n\t\t{event['ResourceStatusReason']}"
```

Comment:
Is it possible that the failure event doesn't has key "ResourceStatusReason"?
In this case, using evet['ResourceStatusReason'] will lead to a key error:
For example:
```
>>> a = {"1": "aaa"}
>>> a["1"]
'aaa'
>>> a["2"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: '2'
>>> a.get("1")
'aaa'
# instead using a.get doesn't get the exception
>>> a.get("2")
>>>
```

Reply:
Fixed - thank you.